### PR TITLE
Updated dependencies using `npm-check-updates`

### PR DESCRIPTION
**Pull request by dantecesa**
Used `npm-check-updates` to bump dependencies to latest versions (mostly because I want to use this without having to use `--legacy-peer-deps` when installing w/ React 18 → see #62).

Note, tests fail locally (see below), but they also fail for the latest on `main`… so not sure what's going on there. They seem to be fine in CI here (so maybe there's some Jest settings 🧙 that I'm missing…).
```
styled-react-modal [dc-updatedependencies] : npm run test

> styled-react-modal@3.0.0 test
> jest --coverage --verbose

 FAIL  tests/Modal.test.jsx
  <Modal />
    ✕ renders nothing when not open (1 ms)
    ✕ renders children when open
    ✕ calls onBackgroundClick when the background is clicked
    ✕ calls onEscapeKeydown when the escape key is pressed
    ✕ calls beforeOpen() before it opens
    ✕ calls beforeOpen() before it opens and waits to call afterOpen() if it returns a promise
    ✕ calls afterOpen() after it opens
    ✕ calls beforeClose() before it closes
    ✕ calls beforeClose() before it closes and waits to call afterClose() if it returns a promise
    ✕ calls afterClose() after it closes (1 ms)
    ✕ passes background props to background
  Modal.styled()
    ✕ returns to a <Modal /> instance (1 ms)

  ● <Modal /> › renders nothing when not open

    The error below may be caused by using the wrong test environment, see https://jestjs.io/docs/configuration#testenvironment-string.
    Consider using the "jsdom" test environment.
    
    ReferenceError: document is not defined

      30 |   };
      31 |
    > 32 |   return render(
         |          ^
      33 |     <ModalProvider data-testid="provider" {...providerProps}>
      34 |       <StatefulModal {...finalModalProps} />
      35 |     </ModalProvider>

      at render (node_modules/@testing-library/react/dist/pure.js:210:5)
      at renderWithProvider (tests/Modal.test.jsx:32:10)
      at Object.renderWithProvider (tests/Modal.test.jsx:41:29)

  ● <Modal /> › renders children when open

    The error below may be caused by using the wrong test environment, see https://jestjs.io/docs/configuration#testenvironment-string.
    Consider using the "jsdom" test environment.
    
    ReferenceError: document is not defined

      30 |   };
      31 |
    > 32 |   return render(
         |          ^
      33 |     <ModalProvider data-testid="provider" {...providerProps}>
      34 |       <StatefulModal {...finalModalProps} />
      35 |     </ModalProvider>

      at render (node_modules/@testing-library/react/dist/pure.js:210:5)
      at renderWithProvider (tests/Modal.test.jsx:32:10)
      at Object.renderWithProvider (tests/Modal.test.jsx:46:27)

  ● <Modal /> › calls onBackgroundClick when the background is clicked

    The error below may be caused by using the wrong test environment, see https://jestjs.io/docs/configuration#testenvironment-string.
    Consider using the "jsdom" test environment.
    
    ReferenceError: document is not defined

      30 |   };
      31 |
    > 32 |   return render(
         |          ^
      33 |     <ModalProvider data-testid="provider" {...providerProps}>
      34 |       <StatefulModal {...finalModalProps} />
      35 |     </ModalProvider>

      at render (node_modules/@testing-library/react/dist/pure.js:210:5)
      at renderWithProvider (tests/Modal.test.jsx:32:10)
      at Object.renderWithProvider (tests/Modal.test.jsx:52:29)

  ● <Modal /> › calls onEscapeKeydown when the escape key is pressed

    The error below may be caused by using the wrong test environment, see https://jestjs.io/docs/configuration#testenvironment-string.
    Consider using the "jsdom" test environment.
    
    ReferenceError: document is not defined

      30 |   };
      31 |
    > 32 |   return render(
         |          ^
      33 |     <ModalProvider data-testid="provider" {...providerProps}>
      34 |       <StatefulModal {...finalModalProps} />
      35 |     </ModalProvider>

      at render (node_modules/@testing-library/react/dist/pure.js:210:5)
      at renderWithProvider (tests/Modal.test.jsx:32:10)
      at Object.renderWithProvider (tests/Modal.test.jsx:63:5)

  ● <Modal /> › calls beforeOpen() before it opens

    The error below may be caused by using the wrong test environment, see https://jestjs.io/docs/configuration#testenvironment-string.
    Consider using the "jsdom" test environment.
    
    ReferenceError: document is not defined

      30 |   };
      31 |
    > 32 |   return render(
         |          ^
      33 |     <ModalProvider data-testid="provider" {...providerProps}>
      34 |       <StatefulModal {...finalModalProps} />
      35 |     </ModalProvider>

      at render (node_modules/@testing-library/react/dist/pure.js:210:5)
      at renderWithProvider (tests/Modal.test.jsx:32:10)
      at Object.renderWithProvider (tests/Modal.test.jsx:74:29)

  ● <Modal /> › calls beforeOpen() before it opens and waits to call afterOpen() if it returns a promise

    The error below may be caused by using the wrong test environment, see https://jestjs.io/docs/configuration#testenvironment-string.
    Consider using the "jsdom" test environment.
    
    ReferenceError: document is not defined

      30 |   };
      31 |
    > 32 |   return render(
         |          ^
      33 |     <ModalProvider data-testid="provider" {...providerProps}>
      34 |       <StatefulModal {...finalModalProps} />
      35 |     </ModalProvider>

      at render (node_modules/@testing-library/react/dist/pure.js:210:5)
      at renderWithProvider (tests/Modal.test.jsx:32:10)
      at renderWithProvider (tests/Modal.test.jsx:89:29)
      at tryCatch (node_modules/@babel/runtime/helpers/regeneratorRuntime.js:86:17)
      at Generator._invoke (node_modules/@babel/runtime/helpers/regeneratorRuntime.js:66:24)
      at Generator.next (node_modules/@babel/runtime/helpers/regeneratorRuntime.js:117:21)
      at asyncGeneratorStep (node_modules/@babel/runtime/helpers/asyncToGenerator.js:3:24)
      at _next (node_modules/@babel/runtime/helpers/asyncToGenerator.js:25:9)
      at node_modules/@babel/runtime/helpers/asyncToGenerator.js:32:7

  ● <Modal /> › calls afterOpen() after it opens

    The error below may be caused by using the wrong test environment, see https://jestjs.io/docs/configuration#testenvironment-string.
    Consider using the "jsdom" test environment.
    
    ReferenceError: document is not defined

      30 |   };
      31 |
    > 32 |   return render(
         |          ^
      33 |     <ModalProvider data-testid="provider" {...providerProps}>
      34 |       <StatefulModal {...finalModalProps} />
      35 |     </ModalProvider>

      at render (node_modules/@testing-library/react/dist/pure.js:210:5)
      at renderWithProvider (tests/Modal.test.jsx:32:10)
      at Object.renderWithProvider (tests/Modal.test.jsx:103:29)

  ● <Modal /> › calls beforeClose() before it closes

    The error below may be caused by using the wrong test environment, see https://jestjs.io/docs/configuration#testenvironment-string.
    Consider using the "jsdom" test environment.
    
    ReferenceError: document is not defined

      30 |   };
      31 |
    > 32 |   return render(
         |          ^
      33 |     <ModalProvider data-testid="provider" {...providerProps}>
      34 |       <StatefulModal {...finalModalProps} />
      35 |     </ModalProvider>

      at render (node_modules/@testing-library/react/dist/pure.js:210:5)
      at renderWithProvider (tests/Modal.test.jsx:32:10)
      at Object.renderWithProvider (tests/Modal.test.jsx:115:29)

  ● <Modal /> › calls beforeClose() before it closes and waits to call afterClose() if it returns a promise

    The error below may be caused by using the wrong test environment, see https://jestjs.io/docs/configuration#testenvironment-string.
    Consider using the "jsdom" test environment.
    
    ReferenceError: document is not defined

      30 |   };
      31 |
    > 32 |   return render(
         |          ^
      33 |     <ModalProvider data-testid="provider" {...providerProps}>
      34 |       <StatefulModal {...finalModalProps} />
      35 |     </ModalProvider>

      at render (node_modules/@testing-library/react/dist/pure.js:210:5)
      at renderWithProvider (tests/Modal.test.jsx:32:10)
      at renderWithProvider (tests/Modal.test.jsx:131:29)
      at tryCatch (node_modules/@babel/runtime/helpers/regeneratorRuntime.js:86:17)
      at Generator._invoke (node_modules/@babel/runtime/helpers/regeneratorRuntime.js:66:24)
      at Generator.next (node_modules/@babel/runtime/helpers/regeneratorRuntime.js:117:21)
      at asyncGeneratorStep (node_modules/@babel/runtime/helpers/asyncToGenerator.js:3:24)
      at _next (node_modules/@babel/runtime/helpers/asyncToGenerator.js:25:9)
      at node_modules/@babel/runtime/helpers/asyncToGenerator.js:32:7

  ● <Modal /> › calls afterClose() after it closes

    The error below may be caused by using the wrong test environment, see https://jestjs.io/docs/configuration#testenvironment-string.
    Consider using the "jsdom" test environment.
    
    ReferenceError: document is not defined

      30 |   };
      31 |
    > 32 |   return render(
         |          ^
      33 |     <ModalProvider data-testid="provider" {...providerProps}>
      34 |       <StatefulModal {...finalModalProps} />
      35 |     </ModalProvider>

      at render (node_modules/@testing-library/react/dist/pure.js:210:5)
      at renderWithProvider (tests/Modal.test.jsx:32:10)
      at Object.renderWithProvider (tests/Modal.test.jsx:146:29)

  ● <Modal /> › passes background props to background

    The error below may be caused by using the wrong test environment, see https://jestjs.io/docs/configuration#testenvironment-string.
    Consider using the "jsdom" test environment.
    
    ReferenceError: document is not defined

      30 |   };
      31 |
    > 32 |   return render(
         |          ^
      33 |     <ModalProvider data-testid="provider" {...providerProps}>
      34 |       <StatefulModal {...finalModalProps} />
      35 |     </ModalProvider>

      at render (node_modules/@testing-library/react/dist/pure.js:210:5)
      at renderWithProvider (tests/Modal.test.jsx:32:10)
      at Object.renderWithProvider (tests/Modal.test.jsx:161:29)

  ● Modal.styled() › returns to a <Modal /> instance

    The error below may be caused by using the wrong test environment, see https://jestjs.io/docs/configuration#testenvironment-string.
    Consider using the "jsdom" test environment.
    
    ReferenceError: document is not defined

      179 |     `;
      180 |
    > 181 |     const { getByTestId } = render(
          |                             ^
      182 |       <ModalProvider>
      183 |         <StyledModal isOpen={true} data-testid="modal" />
      184 |       </ModalProvider>

      at render (node_modules/@testing-library/react/dist/pure.js:210:5)
      at Object.<anonymous> (tests/Modal.test.jsx:181:29)

-------------------|---------|----------|---------|---------|-------------------
File               | % Stmts | % Branch | % Funcs | % Lines | Uncovered Line #s 
-------------------|---------|----------|---------|---------|-------------------
All files          |    12.9 |     2.32 |    5.55 |    12.9 |                   
 Modal.jsx         |      10 |     2.43 |    6.66 |      10 | 8-133,144         
 ModalProvider.jsx |      10 |        0 |       0 |      10 | 10-26             
 baseStyles.jsx    |     100 |      100 |     100 |     100 |                   
 context.jsx       |     100 |      100 |     100 |     100 |                   
 index.jsx         |       0 |        0 |       0 |       0 |                   
-------------------|---------|----------|---------|---------|-------------------
Test Suites: 1 failed, 1 total
Tests:       12 failed, 12 total
Snapshots:   0 total
Time:        0.373 s, estimated 1 s
Ran all test suites.
```

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -34,37 +34,37 @@
   "peerDependencies": {
     "react": "^17.0.0 || ^16.9.0",
```
Comment (AlexanderRichey) [inline package.json:35]: To fix the issue you described in #62 you would have to add `^18.0.0` here.

**Review by AlexanderRichey — COMMENTED** [review]
Thanks for taking the time to submit this PR. I addressed the issue in [v3.0.1](https://github.com/AlexanderRichey/styled-react-modal/releases/tag/v3.0.1). There were some very annoying issues going on with Travis that I had to fix—I ended up switching to CircleCI because of the issues with Travis—which prevented me from merging this PR.

Please let me know if you run into any issues with the latest version and thanks again!

File: package.json (hunk 1 above)
Comment (dantecesa) [reply package.json:35]: Aha!

**Comment by dantecesa** [comment]
Appreciate the follow up & the fix!!

**Pull request closed without merging**